### PR TITLE
SampleRate: Minor fix and improvement

### PR DESCRIPTION
**Pull request by uklotzde**

### CODE REVIEW

File: src/audio/types.h [hunk 1]
```diff
@@ -161,11 +161,15 @@ class SampleRate {
         return value();
     }
 
-    static constexpr SampleRate fromDouble(double value) {
-        return SampleRate(static_cast<value_t>(value));
+    static SampleRate fromDouble(double value) {
+        const auto sampleRate = SampleRate(static_cast<value_t>(value));
+        // The sample rate should always be an integer value
+        // and this conversion is supposed to be lossless.
```
Comment (Holzhaus) [inline src/audio/types.h:167]: good idea. This also catches cases where `value` is NaN?

File: src/audio/types.h [hunk 2]
```diff
@@ -161,11 +161,15 @@ class SampleRate {
...
+        DEBUG_ASSERT(sampleRate.toDouble() == value);
+        return sampleRate;
     }
 
-    constexpr double toDouble() {
+    constexpr double toDouble() const {
```
Comment (Holzhaus) [inline src/audio/types.h:172]: :fearful: oops

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Comment by Holzhaus** [comment]
Waiting for CI.

File: src/audio/types.h (hunk 1 above)
Comment (uklotzde) [reply src/audio/types.h:167]: Not tested. Probably yes, because there is no NaN integer.

**Pull request merged**